toucan_data_sdk/utils/helpers.py

```diff
@@ -1,15 +1,35 @@
 import inspect
 import linecache
+import locale
+import logging
 import re
 import shutil
+import threading
+from contextlib import contextmanager
 from pathlib import Path
-from typing import List
+from typing import List, Optional
 
 from joblib._store_backends import StoreBackendBase
 from slugify import slugify as _slugify
-import logging
 
 logger = logging.getLogger(__name__)
+LOCALE_LOCK = threading.Lock()
+CURRENT_LOCALE = locale.getlocale()
+
+
+@contextmanager
+def setlocale(name: Optional[str]):
+    """
+    Context manager to set a locale ('en', 'fr', 'de', ...)
+    """
+    if name is not None:
+        name = locale.normalize(f'{name}.UTF-8')
+    with LOCALE_LOCK:
+        saved = locale.setlocale(locale.LC_ALL)
```

Comment:
This could raise a `locale.Error` exception if we send an unsupported locale. I think we should fallback to the process one in that case and emit a warning. What do you think? Anyways, we should not crash.

Reply:
Yep ! C'etait prevu qd j'ai vu la CI peter 👍